### PR TITLE
add circular-buffer, part of 48 in 24

### PR DESCRIPTION
**Pull request by kmarker1101**

### CODE REVIEW

**Comment by kmarker1101** [comment]
@BNAndras I am not sure why this is failing CI. Any suggestions?

**Comment by BNAndras** [comment]
I'll take a look later this week and let you know if I see anything amiss.

**Comment by kmarker1101** [comment]
Thanks. Think I have it. Had to (require 'eieio) in test file.

**Comment by kmarker1101** [comment]
Hold up on this. I need to make sure some tests check for a specific error message on some operations, instead of just that it throws an error.

**Comment by kmarker1101** [comment]
good to review now.

File: exercises/practice/circular-buffer/circular-buffer-test.el [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+(load-file "circular-buffer.el")
+
+(defun create-test-buffer (capacity)
+  "Helper function to create a circular buffer with the given capacity."
+  (circular-buffer :capacity capacity))
+
+(ert-deftest reading-empty-buffer-should-fail ()
+  (let ((buf (create-test-buffer 1)))
+    (should (equal (should-error (read-buff buf))
```
Comment (BNAndras) [inline exercises/practice/circular-buffer/circular-buffer-test.el:18]: Do you think it'd be worthwhile to test for a user-defined error like `(should-error (read-buff buff) :type 'empty-buffer)`? If so, we'd need to add stubs for the error symbols in circular-buffer.el.

Comment (kmarker1101) [reply exercises/practice/circular-buffer/circular-buffer-test.el:18]: Agreed. I will look into it.

**Review by BNAndras — APPROVED** [review]

**Comment by BNAndras** [comment]
Looks good. If you rebase the PR to fix the merge conflict, I can merge the PR.

**Comment by kmarker1101** [comment]
should be ready to merge now.

**Pull request merged**